extensions/mssql/src/errorDiagnostics/errorDiagnosticsProvider.ts

```diff
@@ -0,0 +1,96 @@
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { ISqlOpsFeature, SqlOpsDataClient, SqlOpsFeature } from 'dataprotocol-client';
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+import { AppContext } from '../appContext';
+import { ServerCapabilities, ClientCapabilities, RPCMessageType } from 'vscode-languageclient';
+import { Disposable } from 'vscode';
+import * as CoreConstants from '../constants';
+import * as ErrorDiagnosticsConstants from './errorDiagnosticsConstants';
+
+export class ErrorDiagnosticsProvider extends SqlOpsFeature<any> {
+	private static readonly messagesTypes: RPCMessageType[] = [];
+
+	public static asFeature(context: AppContext): ISqlOpsFeature {
+		return class extends ErrorDiagnosticsProvider {
+			constructor(client: SqlOpsDataClient) {
+				super(context, client);
+			}
+
+			override fillClientCapabilities(capabilities: ClientCapabilities): void { }
+
+			override initialize(): void {
+				this.register(this.messages, {
+					id: UUID.generateUuid(),
+					registerOptions: undefined
+				});
+			}
+
+			private convertToIConnectionProfile(profile: azdata.connection.ConnectionProfile): azdata.IConnectionProfile {
+				return {
+					providerName: profile.providerId,
+					id: profile.connectionId,
+					connectionName: profile.connectionName,
+					serverName: profile.serverName,
+					databaseName: profile.databaseName,
+					userName: profile.userName,
+					password: profile.password,
+					authenticationType: profile.authenticationType,
+					savePassword: profile.savePassword,
+					groupFullName: profile.groupFullName,
+					groupId: profile.groupId,
+					saveProfile: profile.savePassword,
+					azureTenantId: profile.azureTenantId,
+					options: profile.options
+				};
+			}
+
+			protected override registerProvider(options: any): Disposable {
+				let handleConnectionError = async (errorCode: number, errorMessage: string, connection: azdata.connection.ConnectionProfile): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> => {
+					if (errorCode === ErrorDiagnosticsConstants.MssqlPasswordResetErrorCode) {
+						// Need to convert inputed profile back to IConnectionProfile.
+						let restoredProfile = this.convertToIConnectionProfile(connection);
+						let result = undefined;
+						try {
+							result = await azdata.connection.openChangePasswordDialog(restoredProfile);
+						}
+						catch (e) {
+							//Error status will be handled below.
+						}
+						if (result === undefined) {
+							return { handled: false };
+						}
+						else {
+							// MSSQL uses 'password' as the option key for connection profile.
+							restoredProfile.options['password'] = result
+							return { handled: true, options: restoredProfile.options };
+						}
```

Comment:
I would also suggest logging an error of some sort if openChangePasswordDialog throws
```suggestion
						catch (e) {
							return { handled: false };
						}
						// MSSQL uses 'password' as the option key for connection profile.
						restoredProfile.options['password'] = result
						return { handled: true, options: restoredProfile.options };
```

Reply:
Turns out this change will change the password to undefined on the connection dialog side if a user was to cancel the password change. To fix this I switched it back to the original design, added some clarifying comments and cleaned it a bit.